### PR TITLE
Aptos gas station client

### PR DESCRIPTION
**Pull request by jasonxh**
Changes:

- Create Aptos gas station client.
- Refactor `@shinami/clients` into `sui` and `aptos` sub-packages.
  - Both `@aptos-labs/ts-sdk` and `@mysten/sui.js` are made optional peer deps.
- Update `eslint` setup and fix lint errors.
- Add `prettier` and fix format errors.

[sc-2855]

### CODE REVIEW

File: packages/nextjs-zklogin/src/client/components/login.tsx
Comment (wnharper) [inline packages/nextjs-zklogin/src/client/components/login.tsx:1]: I'm curious why some of the types in this file were removed?

**Review by wnharper — APPROVED** [review]
🏄🏻‍♂️🏄🏻‍♂️🏄🏻‍♂️

File: packages/nextjs-zklogin/src/client/components/login.tsx
Comment (jasonxh) [reply packages/nextjs-zklogin/src/client/components/login.tsx:1]: It's this `eslint` rule https://typescript-eslint.io/rules/no-inferrable-types/

File: packages/clients/package.json [hunk 1]
```diff
@@ -53,18 +64,28 @@
...
+    "eslint": "^8.57.0",
     "jest": "^29.5.0",
     "ts-jest": "^29.1.0",
     "ts-node": "^10.9.1",
     "typescript": "^5.2.2",
+    "typescript-eslint": "^7.7.1",
     "uuid": "^9.0.0"
   },
   "peerDependencies": {
+    "@aptos-labs/ts-sdk": ">=1.13.1",
```
Comment (aaronTufts) [inline packages/clients/package.json:80]: The Aptos SDK is now on 1.15.0 with a couple breaking changes (not sure if they affect any tests).

The Sui SDK is now on 0.54.1

Comment (jasonxh) [reply packages/clients/package.json:80]: Here we only need to specify the minimum supported versions. I manually tested with the latest Aptos SDK and it should be fine. We have a regular [CI job](https://github.com/shinamicorp/shinami-typescript-sdk/actions/workflows/dep_compat.yaml) to help detect true breaking changes.

**Pull request merged**